packages/color-scale-block/src/ColorScaleBlock.tsx

```diff
@@ -429,81 +398,78 @@ export const ColorScaleBlock: FC<BlockProps> = ({ appBridge }) => {
-                                        ])}
-                                        color={color}
-                                        onDrop={handleDrop}
-                                        onDelete={handleColorSquareDelete}
-                                        onResizeStart={handleResizeStart}
-                                        canDragAndDrop={!resizedColorIndex.current}
-                                        isEditing={isEditing}
-                                        setCurrentlyDraggedColorId={setCurrentlyDraggedColorId}
-                                        currentlyDraggedColorId={currentlyDraggedColorId}
-                                    />
-                                </div>
-                            );
-                        })}
-
-                        {displayableItems.length === 0 && (
-                            <EmptyView
-                                height={
-                                    blockSettings.customHeight ? blockSettings.heightInput : blockSettings.heightSlider
-                                }
-                            />
-                        )}
-                    </DndProvider>
-                </div>
+                {displayableItems.length > 0 && (
+                    <div
+                        style={{
+                            height: colorScaleHeight,
+                        }}
+                        className="tw-rounded tw-w-full tw-flex"
+                        // Note: onMouseUp and handleResize are defined here intentionally, instead of being in the DragHandle component.
+                        // The reason for this is that the dragging feature stops working if I move these to DragHandle,
+                        // perhaps because the component is being destroyed on every re-render and causing issues with dragging.
+                        // The 'handleResizeStart' method, on the other hand, needs to stay in DragHandle, because it needs to
+                        // identify which color square is being resized.
+                    >
+                        <DndProvider backend={HTML5Backend}>
+                            {displayableItems.map((color: ColorProps, index: number) => {
+                                const width =
+                                    currentlyDraggedColorId === color.id
+                                        ? 0
+                                        : color.width ??
+                                          calculateDefaultColorWidth(displayableItems.length, colorScaleBlockRef);
+                                const isFirst = index === 0;
+                                const isLast = index === displayableItems.length - 1;
+
+                                return (
+                                    <div
+                                        data-test-id="color-wrapper"
+                                        className="tw-pr-[1px] tw-flex tw-group tw-relative tw-h-full"
```

Comment:
you can remove here the 'tw-pr-[1px]' and add `tw-justify-between` to line 414 to make remove the right-side gap issue

Reply:
or you can just remove it here and add it to the ColorSquare where you have the bg color. No need for justify between.